ipc/provider/src/manager/evm/manager.rs

```diff
@@ -1777,36 +1775,42 @@ impl TryFrom<gateway_getter_facet::Subnet> for SubnetInfo {
     }
 }
 
-fn convert_msg(
-    msg: checkpointing_facet::IpcEnvelope,
-) -> subnet_actor_checkpointing_facet::IpcEnvelope {
-    subnet_actor_checkpointing_facet::IpcEnvelope {
-        kind: msg.kind,
-        local_nonce: msg.local_nonce,
-        from: subnet_actor_checkpointing_facet::Ipcaddress {
-            subnet_id: subnet_actor_checkpointing_facet::SubnetID {
-                root: msg.from.subnet_id.root,
-                route: msg.from.subnet_id.route,
-            },
-            raw_address: subnet_actor_checkpointing_facet::FvmAddress {
-                addr_type: msg.from.raw_address.addr_type,
-                payload: msg.from.raw_address.payload,
-            },
-        },
-        to: subnet_actor_checkpointing_facet::Ipcaddress {
-            subnet_id: subnet_actor_checkpointing_facet::SubnetID {
-                root: msg.to.subnet_id.root,
-                route: msg.to.subnet_id.route,
-            },
-            raw_address: subnet_actor_checkpointing_facet::FvmAddress {
-                addr_type: msg.to.raw_address.addr_type,
-                payload: msg.to.raw_address.payload,
-            },
+const SIGNED_MSG_TYPE_PRECOMMIT: u8 = 2;
+
+fn convert_signed_header(signed_header: SignedHeader) -> ethers::abi::Bytes {
+    let light_header = signed_header.header;
+
+    let vote_template = CanonicalVoteData {
+        vote_type: SIGNED_MSG_TYPE_PRECOMMIT,
```

Comment:
Is it guaranteed that CometBFT won't produce other vote types? If so, they need to be skipped.

Reply:
good call, it was done onchain before, I added the filter.